native/membrane_videocompositor/src/elixir_bridge/scene_validation.rs

```diff
@@ -0,0 +1,392 @@
+use std::collections::{HashMap, HashSet};
+
+use super::elixir_structs::*;
+
+impl Object {
+    fn mentioned_names(&self) -> Vec<&Name> {
+        match self {
+            Object::Layout(layout) => layout.mentioned_names(),
+            Object::Texture(texture) => texture.mentioned_names(),
+            Object::Video(_) => Vec::new(),
+        }
+    }
+}
+
+impl Texture {
+    fn mentioned_names(&self) -> Vec<&Name> {
+        let mut names = vec![&self.input];
+
+        if let TextureOutputResolution::Name(name) = &self.resolution {
+            names.push(name);
+        }
+
+        names
+    }
+}
+
+impl Layout {
+    fn mentioned_names(&self) -> Vec<&Name> {
+        let mut names = self.inputs.values().collect::<Vec<_>>();
+
+        if let LayoutOutputResolution::Name(name) = &self.resolution {
+            names.push(name);
+        }
+
+        names
+    }
+}
+
+impl Scene {
+    fn check_for_duplicate_pad_refs(
+        objects: &[(ObjectName, Object)],
+    ) -> Result<(), SceneParsingError> {
+        let mut set = HashSet::new();
+
+        for (_, object) in objects {
+            if let Object::Video(InputVideo { input_pad }) = object {
+                if set.contains(input_pad) {
+                    return Err(SceneParsingError::DuplicatePadReferences(input_pad.clone()));
+                }
+
+                set.insert(input_pad);
+            }
+        }
+
+        Ok(())
+    }
```

Comment:
[NIT] The `Set` name is too vague, it doesn't help to understand the code. Use `input_pads` or a similar name for this.